File: direct/1C.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        var scanner = new Scanner(System.in);
        var poles = new Pos[3];
        for (int i = 0; i < 3; i++)
            poles[i] = Pos.fromScanner(scanner);

        var circle = circleFromPoints(poles[0], poles[1], poles[2]);

        double area;
        outer: for (var sides = 3;; sides++) {
            var pairs = new Pos[][] { { poles[0], poles[1] }, { poles[1], poles[2] }, { poles[2], poles[0] } };
            for (var pair : pairs) {
                var angleA = Math.atan2(pair[0].y - circle.center.y, pair[0].x - circle.center.x);
                var angleB = Math.atan2(pair[1].y - circle.center.y, pair[1].x - circle.center.x);
                var angle = Math.abs(angleA - angleB);

                if (Math.abs(Math.sin(sides * angle / 2)) > 1e-3)
                    continue outer;
            }

            area = (square(circle.radius) * sides * Math.sin((2 * Math.PI) / sides)) / 2;
            break;
        }

        System.out.printf("%.6f\n", area);
    }

    static double square(double x) {
        return x * x;
    }

    static Circle circleFromPoints(Pos a, Pos b, Pos c) {
        var calcA = a.x * (b.y - c.y) - a.y * (b.x - c.x) + b.x * c.y - c.x * b.y;
        var calcB = (Math.pow(a.x, 2) + Math.pow(a.y, 2)) * (c.y - b.y) +
                (Math.pow(b.x, 2) + Math.pow(b.y, 2)) * (a.y - c.y) +
                (Math.pow(c.x, 2) + Math.pow(c.y, 2)) * (b.y - a.y);
        var calcC = (Math.pow(a.x, 2) + Math.pow(a.y, 2)) * (b.x - c.x) +
                (Math.pow(b.x, 2) + Math.pow(b.y, 2)) * (c.x - a.x) +
                (Math.pow(c.x, 2) + Math.pow(c.y, 2)) * (a.x - b.x);
        var calcD = (Math.pow(a.x, 2) + Math.pow(a.y, 2)) * (c.x * b.y - b.x * c.y) +
                (Math.pow(b.x, 2) + Math.pow(b.y, 2)) * (a.x * c.y - c.x * a.y) +
                (Math.pow(c.x, 2) + Math.pow(c.y, 2)) * (b.x * a.y - a.x * b.y);

        var x = -calcB / (2 * calcA);
        var y = -calcC / (2 * calcA);
        var radius = Math.sqrt(Math.pow(calcB, 2) + Math.pow(calcC, 2) - 4 * calcA * calcD) / (2 * Math.abs(calcA));

        return new Circle(new Pos(x, y), radius);
    }

    static class Circle {
        Pos center;
        double radius;

        Circle(Pos center, double radius) {
            this.center = center;
            this.radius = radius;
        }
    }

    static class Pos {
        double x;
        double y;

        Pos(double x, double y) {
            this.x = x;
            this.y = y;
        }

        static Pos fromScanner(Scanner scanner) {
            return new Pos(scanner.nextDouble(), scanner.nextDouble());
        }

        double dist(Pos other) {
            return Math.sqrt(Math.pow(x - other.x, 2) + Math.pow(y - other.y, 2));
        }
    }
}

File: direct/1C.kt
import java.util.Scanner
import kotlin.math.*

class Vector2D(val x: Double, val y: Double) {
    val magnitude: Double
        get() {
            return sqrt(x*x+y*y)
        }

    val angle: Double
        get() {
            return atan2(y, x)
        }

    fun subtract(other: Vector2D): Vector2D {
        return Vector2D(
                x - other.x,
                y - other.y
        )
    }

    fun angleTo(other: Vector2D): Double {
        return other.angle - angle
    }

    override fun toString(): String {
        return "<${x}, ${y}>"
    }
}

val EPS = 0.0001

fun main() {
    val scan = Scanner(System.`in`)

    val vertices = listOf(
        Vector2D(scan.nextDouble(), scan.nextDouble()),
        Vector2D(scan.nextDouble(), scan.nextDouble()),
        Vector2D(scan.nextDouble(), scan.nextDouble())
    )

    for (n in 3..100) {
        // Check to see if the given vertices could represent vertices of an
        // n-gon
        val a = vertices[1].x*vertices[1].x - vertices[0].x*vertices[0].x + vertices[1].y*vertices[1].y - vertices[0].y*vertices[0].y
        val b = vertices[2].x*vertices[2].x - vertices[0].x*vertices[0].x + vertices[2].y*vertices[2].y - vertices[0].y*vertices[0].y
        val c = 2*vertices[1].x - 2*vertices[0].x
        val d = 2*vertices[2].x - 2*vertices[0].x
        val e = 2*vertices[0].y - 2*vertices[1].y
        val f = 2*vertices[0].y - 2*vertices[2].y

        val center = Vector2D(
                (a+e*((a*d-b*c)/(f*c-e*d)))/c,
                (a*d-b*c)/(f*c-e*d)
        )
        //println("Trying n ${n} with center ${center} and radius ${vertices[0].subtract(center).magnitude}")
        var pass = true
        for (i in 0..<vertices.size) {
            // The angle between all pairs should be a multiple of 360deg / n (or 2*pi/n)
            val angb = vertices[i].subtract(center).angleTo(vertices[if (i < vertices.size-1) i+1 else 0].subtract(center))
            //println(angb)
            if (abs(angb) < (2*Math.PI/n - 1e-7) || abs(angb) % (2*Math.PI/n - 1e-7) > EPS) {
                pass = false
                break
            }
        }
        if (pass) {
            val o = vertices[0].subtract(center).magnitude * sin(Math.PI / n)
            val a = vertices[0].subtract(center).magnitude * cos(Math.PI / n)
            val segmentArea = o*a
            val totalArea = segmentArea*n
            println(totalArea)
            return
        }
    }
    println("No ans for ${vertices}!")
}